integration-tests/tests/common/mod.rs

```diff
@@ -76,17 +84,30 @@ impl IntegrationTest {
             .arg(self.code)
             .arg("--shim")
             .arg(self.shim)
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
             .spawn()
             .expect("failed to run the test");
 
+        cmd.stdin
+            .as_mut()
+            .unwrap()
+            .write_all(stdin.as_ref())
+            .expect("Failed to write stdin");
```

Comment:
Is this being used for the `write()` test? If so, I'm not understanding the test correctly.

Reply:
No wouldn't need stdin for the write test, but we would need these changes for read.
Should I remove them?

Reply:
I would probably opt for only running that if there is a `stdin` that isn't empty. @npmccallum were you suggesting that this should be run every time even without a `stdin` argument?

Reply:
I'm okay to leave this. `write()` with a zero-length input is a noop.